### ISSUE TITLE
[Bug]: AttributeError: 'NoneType' object has no attribute 'llm'

### ISSUE THREAD

**Issue by chowgi** [run-llama/llama_index#10633]
### Bug Description

When I change from using as_query_engine to as_chat_engine using the new method with no Service Context I get he following.

AttributeError: 'NoneType' object has no attribute 'llm'

### Version

0.10

### Steps to Reproduce

**This code works fine**
query_engine = index.as_query_engine(streaming=True)
response = query_engine.query("What did the author do growing up?")

This code results in the **error**
query_engine = index.as_chat_engine()
response = query_engine.chat("What did the author do growing up?")

### Relevant Logs/Tracbacks

```shell
Traceback (most recent call last):
  File "/home/runner/LllamaIndex/main.py", line 23, in <module>
    query_engine = index.as_chat_engine()
  File "/home/runner/LllamaIndex/.pythonlibs/lib/python3.10/site-packages/llama_index/core/indices/base.py", line 423, in as_chat_engine
    llm = service_context.llm
AttributeError: 'NoneType' object has no attribute 'llm'
```

**Comment by anoopshrma** [comment]
It should by default set a mode if not passed though

**Comment by logan-markewich** [comment]
@chowgi this should be fixed, I will be releasing a new version shortly

**Comment by FaysalEtr** [comment]
# get LLM
            # service_context = cast(ServiceContext, kwargs["service_context"])
            # llm = service_context.llm   
Commenting out those two lines of code worked for me.

**Comment by logan-markewich** [comment]
This is fixed in v0.10.3 of llama-index (and llama-index-core)

`pip install -U llama-index llama-index-core`

**Comment by chowgi** [comment]
> This is fixed in v0.10.3 of llama-index (and llama-index-core)
> 
> `pip install -U llama-index llama-index-core`

Thanks!

**Comment by casualcomputer** [comment]
Still have the same issue
